internal/app/cache/cache.go

```diff
@@ -19,10 +19,13 @@ type Cache interface {
 	Fetch(key string) (*Resource, error)
 
 	// SetResponse sets the cache response and returns the list of requests.
-	SetResponse(key string, resp v2.DiscoveryResponse) ([]*v2.DiscoveryRequest, error)
+	SetResponse(key string, resp v2.DiscoveryResponse) (map[*v2.DiscoveryRequest]bool, error)
```

Comment:
Just a thought ...
Alternative is to keep this interface and put the map into the [internal cache struct](https://github.com/envoyproxy/xds-relay/pull/59/files#diff-daf2807d54a5369eb2a29014b0d051cfR31). An advantage of keeping the slice is ordered keys when orchestrator or other callers wants to test the results against a mocked Response. But sorting adds an overhead to SetResponse. Since orchestrator doesn't care about order during fanout, this is probably ok or preferable for now.